dll/win32/shell32/CDefView.cpp

```diff
@@ -2182,6 +2198,15 @@ LRESULT CDefView::OnContextMenu(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &b
     if (FAILED_UNEXPECTEDLY(hResult))
         return 0;
 
+    if (m_pCommDlgBrowser && !(GetCommDlgViewFlags() & CDB2GVF_NOSELECTVERB))
+    {
+        HMENU hMenuSource = LoadMenu(_AtlBaseModule.GetResourceInstance(), MAKEINTRESOURCEW(IDM_DVSELECT));
+        Shell_MergeMenus(m_hContextMenu, GetSubMenu(hMenuSource, 0), 0, DVIDM_COMMDLG_SELECT, 0xffff, MM_ADDSEPARATOR | MM_DONTREMOVESEPS);
```

Comment:
Why the MM_DONTREMOVESEPS` ?

Reply:
Because I don't want to mess with whatever `QueryContextMenu` (a shell extension we know nothing about) did to the menu. If they want double separators, that is not my business, I'm only adding to the menu, not cleaning it up.

Reply:
understandable